### PR TITLE
Removed ZooKeeper dependency and used Admin Client for user quotas

### PR DESCRIPTION
**Pull request by ppatierno**
### Type of change

- Refactoring

### Description

This PR fixes https://github.com/strimzi/strimzi-kafka-operator/issues/3112 actually removing the ZooKeeper dependency from the `KafkaUserQuotasOperator` and using the Admin Client API instead. 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging

### CODE REVIEW

**Comment by scholzj** [comment]
This can be probably merged only after the support for Kafka 2.5.x is removed after branching 0.21, or?

**Comment by ppatierno** [comment]
Right. I opened this issue https://github.com/strimzi/strimzi-kafka-operator/issues/4224 to work on for it.

**Comment by ppatierno** [comment]
This PR can be merged only after #4491 is merged.

File: user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotasOperator.java [hunk 1]
```diff
@@ -202,119 +92,105 @@ boolean isPathExist(String path)    {
...
-            }
-            if (config.getString("request_percentage") != null) {
-                config.remove("request_percentage");
+    protected KafkaUserQuotas describeUserQuotas(String username) throws Exception {
+        ClientQuotaFilterComponent c = ClientQuotaFilterComponent.ofEntity(ClientQuotaEntity.USER, username);
+        ClientQuotaFilter f =  ClientQuotaFilter.contains(Collections.singleton(c));
+        KafkaUserQuotas current = null;
+        try {
+            ClientQuotaEntity cqe = new ClientQuotaEntity(Collections.singletonMap(ClientQuotaEntity.USER, username));
+            Map<ClientQuotaEntity, Map<String, Double>> map = adminClient.describeClientQuotas(f).entities().get();
```
Comment (scholzj) [inline user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotasOperator.java:136]: Should this be done asynchronously? Or is this whole block in some `executeBlocking`?

File: user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotasOperator.java [hunk 2]
```diff
@@ -202,119 +92,105 @@ boolean isPathExist(String path)    {
...
-     *
-     * @return  Returns the updated JSON without the quotas
-     */
-    protected JsonObject removeQuotasFromJsonUser(byte[] userConfig)   {
-        JsonObject json = new JsonObject(new String(userConfig, StandardCharsets.UTF_8));
+    protected void alterUserQuotas(String username, Set<ClientQuotaAlteration.Op> ops) throws Exception {
+        ClientQuotaEntity cqe = new ClientQuotaEntity(Collections.singletonMap(ClientQuotaEntity.USER, username));
+        ClientQuotaAlteration cqa = new ClientQuotaAlteration(cqe, ops);
+        try {
+            adminClient.alterClientQuotas(Collections.singleton(cqa)).all().get();
```
Comment (scholzj) [inline user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotasOperator.java:123]: Should this be done asynchronously? Or is this whole block in some `executeBlocking`?

**Review by scholzj — COMMENTED** [review]
I left few comments. I also wonder if you tried whether it works well in combination with the SCRAM-SHA users? I think that in the past we had problem with it because they are stored in the same path in ZK.

File: user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotasOperator.java (hunk 1 above)
Comment (ppatierno) [reply user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotasOperator.java:136]: Yes, all the adminClient calls are done from inside the `reconcile` method (which calls `createOrUpdate`) and the `reconcile` does these calls in an `executeBlocking`. The same happens for the `SimpleAclOperator` for example.

File: user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotasOperator.java (hunk 2 above)
Comment (ppatierno) [reply user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotasOperator.java:123]: Ditto as above.

**Review by scholzj — APPROVED** [review]

**Pull request merged**